### PR TITLE
Add support for samplingRate

### PR DESCRIPTION
**Pull request by suniltheta**
**Issue**
https://github.com/aws/aws-app-mesh-roadmap/issues/371

**Description of changes:**
Add back support for samplingRate
This reverts commit 413e594a3f3f85988768e831a4a7bdbf3b409869. #522

This support was originally added in #512

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/inject/config.go [hunk 1]
```diff
@@ -153,6 +155,8 @@ func (cfg *Config) BindFlags(fs *pflag.FlagSet) {
 		"Enable Envoy X-Ray tracing integration and injects xray-daemon as sidecar")
 	fs.Int32Var(&cfg.XrayDaemonPort, flagXrayDaemonPort, 2000,
 		"X-Ray Agent tracing port")
+	fs.StringVar(&cfg.XraySamplingRate, flagXraySamplingRate, "",
```
Comment (cgchinmay) [inline pkg/inject/config.go:158]: shouldn't the default value be 0.05 ?

Comment (suniltheta) [reply pkg/inject/config.go]: true. Now set the default value 0.05.

File: pkg/inject/sidecar_builder.go [hunk 2]
```diff
@@ -81,6 +84,29 @@ func updateEnvMapForEnvoy(vars EnvoyTemplateVariables, env map[string]string, vn
...
+		samplingRate, ok := env["XRAY_SAMPLING_RATE"]
+		if ok {
+			// `podAnnotations` contains the sampling rate and gets preference over helm configuration
+			// For now delete this value from env so that we can validate before adding again
+			delete(env, "XRAY_SAMPLING_RATE")
+		} else {
+			// `podAnnotations` doesn't contain the sampling rate so get value from helm configuration
+			samplingRate = vars.XraySamplingRate
+		}
+		if samplingRate != "" {
```
Comment (cgchinmay) [inline pkg/inject/sidecar_builder.go:98]: Do we need this check ? The samplingRate can be empty if the customer has erroneously specified "" value in pod annotations. In that case, we can notify the Customer of this error ?

Comment (suniltheta) [reply pkg/inject/sidecar_builder.go:98]: yes we don't need this check. Now that we implicitly always set this to 0.05 and like you mentioned if the user makes a mistake then we notify via an error.

**Review by cgchinmay — APPROVED** [review]
/lgtm

**Comment by cgchinmay** [comment]
@suniltheta What is the behavior with envoy image older than 1.19.1.1 ?

**Comment by suniltheta** [comment]
Envoy image prior to 1.19.1.1, there will be no effect even if we set this value. The X-ray sampling will still pick default value of 0.05.

**Review by cgchinmay — APPROVED** [review]
/lgtm

**Pull request merged**